src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseData/IntellisenseData.cs

```diff
@@ -265,7 +270,73 @@ internal virtual void BeforeAddSuggestionsForDottedNameNode(TexlNode node)
         /// <returns>
         /// True if the function may be suggested, false otherwise.
         /// </returns>
-        internal virtual bool ShouldSuggestFunction(TexlFunction function) => true;
+        internal virtual bool ShouldSuggestFunction(TexlFunction function)
+        {
+            if (CurFunc == null)
+            {
+                return true;
+            }
+
+            // Expected return type is, type of argument at current index, e.g. Collect(| the first arg is the table type, so we expect to suggest all function that can return a table.
+            var expectedReturnType = GetCurrentArgType();
+            var functionReturnType = function.ReturnType;
+
+            if (expectedReturnType.IsTableNonObjNull)
+            {
+                // this avoids non scope tabular functions from being suggested, e.g. Abs([1,-2]) is tabular overload but doesn't have scope.
```

Comment:
>/ this avoids non scope tabular functions from bein [](http://example.com/codeflow?start=17&length=51)

Does that mean that intellisense won't suggest Abs at

Filter(|  ?


Filter(Abs[1,2,3] .. is still valid though.